### PR TITLE
Create wrapper class to make documentation more accessable

### LINKED ISSUE #94: Get Godot Editor to recognize SQLite as a class
**Is your feature request related to a problem? Please describe.**

As I am developing, I often use the built-in help tools such as autocomplete and help text to make sure I am calling the right functions. For this plugin, Godot does not recognize the class, which means I am more likely to mistype functions and need to keep the github documentation up in order to read it.

**Describe the solution you'd like**

I am not familiar with how this plugin was built, so the best I can think of is enclosing the SQLIte component in a wrapper class designed primarily to provide documentation. The user would then load the wrapper class, and gain access to autocomplete functionality and in-ide documentation.

**Describe alternatives you've considered**

I am currently leaving the GitHub page up  while I am developing so I can reference its documentation.

**Additional context**

OS: Linux Mint (X11)
Godot Version: v3.5.1.stable.official
image

### PR DESCRIPTION
**Pull request by engineer-pearl**
I've been careful not to take too much liberty with the documentation, but I added a little extra where I thought it was needed (like the insert_rows function). Someone with a background in technical documentation probably could have done better.

Functionality wise, clicking on functions brings you to where they're defined, and godot can auto-complete function names. I'd say this means I've accomplished the two biggest goals for this helper class.

I'm not too happy with the file name, file location, or class name though, and am open to suggestions if you've got them. 

Do you think the demo project you've written is a sufficient test to ensure the class is feature complete, or are there additional test cases you would recommend I add?

Issue tags:
fixes #94

### CODE REVIEW

**Comment by 2shady4u** [comment]
Hi @engineer-pearl!

Would it be possible to squash your commits into one?
I'll try to properly review your pull request as soon as possible!

**Comment by engineer-pearl** [comment]
There is a mechanism to squash on pull request: https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/configuring-commit-squashing-for-pull-requests

Perhaps we could use that?

**Comment by 2shady4u** [comment]
> There is a mechanism to squash on pull request: https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/configuring-commit-squashing-for-pull-requests
> 
> Perhaps we could use that?

It's much to review a PR for me if the comments are squashed :)

File: demo/addons/godot-sqlite/godot-sqlite-with-help.gd [hunk 1]
```diff
@@ -0,0 +1,299 @@
+class_name SQLite_Documented
+## Wrapper for SQLite class [br]
+## [br]
+## Makes documentation accessable. Can be used almost interchangablt with gdsqlite.gdns [br]
```
Comment (leprasmurf) [inline demo/addons/godot-sqlite/godot-sqlite-with-help.gd:4]: \* interchangeably

File: demo/addons/godot-sqlite/godot-sqlite-with-help.gd [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+	
+	
+## Passes a query as-is to the database
+func query(query_string : String) -> bool:
+	return db.query(query_string)
+
+
+##    Binds the parameters contained in the `param_bindings`-variable to the query. Using this function stops any possible attempts at SQL data injection as the parameters are sanitized. More information regarding parameter bindings can be found [here](https://www.sqlite.org/c3ref/bind_blob.html).
+##
+##    **Example usage**:
```
Comment (leprasmurf) [inline demo/addons/godot-sqlite/godot-sqlite-with-help.gd:173]: Examples are great!  Might be beneficial to show the action on the results.  e.g.,
```gdscript
...
var success = db.query_with_bindings(query_string, param_bindings)

if success:
    for result in db.query_result:
        print result
```

File: demo/addons/godot-sqlite/godot-sqlite-with-help.gd [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+	if db:
+		db.foreign_keys = value
+
+func get_foreign_keys() -> bool:
+	if !db:
+		return false
+	return db.foreign_keys
+#- **read_only** (Boolean, default=false)
+#
+## Enabling this property opens the database in read-only modus & allows databases to be packaged inside of the PCK. [br]
```
[the remark this answers is not in the document]
Comment (engineer-pearl) [reply demo/addons/godot-sqlite/godot-sqlite-with-help.gd:77]: I will update the code to have this change, but this is also an issue in the ReadMe. (Note: I'm considering the readme to be out-of-scope for this change)

File: demo/addons/godot-sqlite/godot-sqlite-with-help.gd (hunk 2 above)
Comment (engineer-pearl) [reply demo/addons/godot-sqlite/godot-sqlite-with-help.gd:173]: That is a good call - I've been to the readme multiple times to check the formatting of the query_result value.

File: demo/addons/godot-sqlite/godot-sqlite-with-help.gd (hunk 1 above)
Comment (engineer-pearl) [reply demo/addons/godot-sqlite/godot-sqlite-with-help.gd]: Updated.

File: demo/addons/godot-sqlite/godot-sqlite-with-help.gd [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+##
+##    ```gdscript
+##    var query_string : String = "SELECT ? FROM company WHERE age < ?;"
+##    var param_bindings : Array = ["name", 24]
+##    var success = db.query_with_bindings(query_string, param_bindings)
+##    # Executes following query: 
+##    # SELECT name FROM company WHERE age < 24;
+##    ```
+##
+##    Using bindings is optional, except for PoolByteArray (= raw binary data) which has to binded to allow the insertion and selection of BLOB data in the database.
```
[the remark this answers is not in the document]
Comment (engineer-pearl) [reply demo/addons/godot-sqlite/godot-sqlite-with-help.gd]: Fixed

**Comment by engineer-pearl** [comment]
@2shady4u the squashed version of this pull request is now outdated. Sorry.

**Comment by engineer-pearl** [comment]
@leprasmurf How are the changes that I made? Can I resolve the conversations?

File: demo/addons/godot-sqlite/godot-sqlite-with-help.gd (hunk 4 above)
[the remark this answers is not in the document]
Comment (leprasmurf) [reply demo/addons/godot-sqlite/godot-sqlite-with-help.gd]: LGTM

[the remark this answers is not in the document]
Comment (engineer-pearl) [reply demo/addons/godot-sqlite/godot-sqlite-with-help.gd]: I am not familiar with that acronym, sorry

[the remark this answers is not in the document]
Comment (engineer-pearl) [reply demo/addons/godot-sqlite/godot-sqlite-with-help.gd]: Internet says it's "Looks good to me" so I'm resolving this.

**Comment by engineer-pearl** [comment]
@2shady4u Leprasmurf's comments have been resolved.

File: demo/addons/godot-sqlite/godot-sqlite-with-help.gd (hunk 4 above)
[the remark this answers is not in the document]
Comment (leprasmurf) [reply demo/addons/godot-sqlite/godot-sqlite-with-help.gd]: yeah, Looks Good To Me is what I intended.

**Comment by 2shady4u** [comment]
Hi @engineer-pearl 

Sorry that it took so long :/
I've finally found some time to review, update and merge this PR.

Thank you for your contribution 🙏

**Pull request merged**